Application/APISupport.js

```diff
@@ -0,0 +1,85 @@
+/* global Environment:false */
+
+'use strict';
+
+angular.module('shared')  
+.factory('APISupport', ['$rootScope', '$http', '$cookies', function($rootScope, $http, $cookies) {
+  // use alternate API URL (mainApiUrl) if provided, otherwise use primary API URL (apiUrl)
+  var apiUrl = Environment.getConfig('mainApiUrl') ? Environment.getConfig('mainApiUrl') : Environment.getConfig('apiUrl');
```

Comment:
Can be rewritten as
`var apiUrl = Environment.getConfig('mainApiUrl') || Environment.getConfig('apiUrl');`